pipeline/runtime/step_executor.go

```diff
@@ -110,7 +113,122 @@ func (e *StepExecutor) PollStep(ctx context.Context, r *api.PollStepRequest) (*a
 	return convertStatus(status), nil
 }
 
+func (e *StepExecutor) StreamOutput(ctx context.Context, r *api.StreamOutputRequest) (oldOut []byte, newOut <-chan []byte, err error) {
```

Comment:
StreamOutput method does not below to this file since it is not related to step execution but for step log streaming. Can you move this method out of this file?

Reply:
`StreamOutput` method is there and should remain there because it's a method of `StepExecutor` type. The struct `StepExecutor` itself and all other its methods are in the same file. Why would we want to move this one to a different file? Besides, a step's output is closely related to its execution, so that's another reason for it to stay where it is. The actual streaming is in an another file: step_output.go

Reply:
Coupling of log streaming handler and step execution is not a good idea